### PR TITLE
sysvar: Do not allow set tidb_auto_analyze_ratio to 0

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/51582

Problem Summary:

Currently, if you set [tidb_auto_analyze_ratio](https://docs.pingcap.com/tidb/stable/system-variables#tidb_auto_analyze_ratio) to 0, we will disable auto-analysis based on row changes, but we will still process the newly added indexes and tables.

This behavior is difficult to understand. I suppose most users will think 0 means we always need to analyze the table, no matter how many rows have been updated.

The historic reason we introduced it is that before https://github.com/pingcap/tidb/pull/34643 this PR, there was no way to disable the auto-analyze feature without restarting the server. So we chose this special value to allow users to disable the auto-analysis feature. But now enable_tidb_auto_analyze is a global session variable. So we don't depend on this value.

### What changed and how does it work?

Added verification for tidb_auto_analyze_ratio to make sure it cannot be zero and the minimum value is 0.00001.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
不允许将 tidb_auto_analyze_ratio 参数设置为0
Do not allow set tidb_auto_analyze_ratio to 0
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
/hold

Need to test the upgrade case.

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
Test locally:
1. start the v7.5.0 tidb cluser
```console
tiup cluster deploy upstream v7.5.0 upstream.yaml -p
tiup cluster start upstream
```
2. Check the version and set the ratio to 0
```sql
MySQL [(none)]> select tidb_version();
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v7.5.0
Edition: Community
Git Commit Hash: 069631e2ecfedc000ffb92c67207bea81380f020
Git Branch: heads/refs/tags/v7.5.0
UTC Build Time: 2023-11-24 08:50:14
GoVersion: go1.21.3
Race Enabled: false
Check Table Before Drop: false
Store: tikv |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

MySQL [(none)]> set global tidb_auto_analyze_ratio=0;
Query OK, 0 rows affected (0.02 sec)

MySQL [(none)]> select @@tidb_auto_analyze_ratio;
+---------------------------+
| @@tidb_auto_analyze_ratio |
+---------------------------+
| 0                         |
+---------------------------+
1 row in set (0.00 sec)

MySQL [(none)]> exit
```
3. Upgrade the tidb: `tiup cluster patch upstream ../../../tidb/bin/tidb-hotfix-linux-amd64.tar.gz -N 10.2.6.x:4000,10.2.6.x:4000,10.2.7.x:4000 -R tidb`
4. Check the version and value:
```sql
MySQL [(none)]> select tidb_version();
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                               |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v8.0.0-alpha-664-gb2bd0b4fbf
Edition: Community
Git Commit Hash: b2bd0b4fbfda97e7f0a698baffc9c1269a84fb6c
Git Branch: rustin-patch-0
UTC Build Time: 2024-03-28 09:04:11
GoVersion: go1.21.1
Race Enabled: false
Check Table Before Drop: false
Store: tikv |
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

MySQL [(none)]> select @@tidb_auto_analyze_ratio;
+---------------------------+
| @@tidb_auto_analyze_ratio |
+---------------------------+
| 0                         |
+---------------------------+
1 row in set (0.00 sec)

MySQL [(none)]> set global tidb_auto_analyze_ratio=0;
ERROR 1105 (HY000): the value of tidb_auto_analyze_ratio should be greater than or equal to 0.000010
MySQL [(none)]> 
```

File: pkg/statistics/handle/autoanalyze/refresher/refresher_test.go [hunk 1]
```diff
@@ -54,8 +54,10 @@ func TestSkipAnalyzeTableWhenAutoAnalyzeRatioIsZero(t *testing.T) {
 	)
 	tk.MustExec("insert into t1 values (1, 1), (2, 2), (3, 3)")
 	tk.MustExec("insert into t2 values (1, 1), (2, 2), (3, 3)")
-	// Set the auto analyze ratio to 0.
-	tk.MustExec("set global tidb_auto_analyze_ratio = 0")
+	// HACK: Set the auto analyze ratio to 0.
```
Comment (hi-rustin) [inline pkg/statistics/handle/autoanalyze/refresher/refresher_test.go:57]: It didn't change the old behavior.

**Review by time-and-fate — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Review by songrijie — APPROVED** [review]

**Pull request merged**